assets/js/components/ImgDropAndCrop/index.js

```diff
@@ -0,0 +1,122 @@
+import React, {Component} from 'react'
+import Dropzone from 'react-dropzone'
+/*eslint-disable */
+import ReactCrop from 'react-image-crop'
+/*eslint-enable */
+
+import '../../../../node_modules/react-image-crop/dist/ReactCrop.css';
+import {image64toCanvasRef, extractImageFileExtensionFromBase64} from './ReusableUtils';
+
+const imageMaxSize = 100000000
+const acceptedFileTypes = 'image/x-png, image/png, image/jpg, image/jpeg, image/gif'
+const acceptedFileTypesArray = acceptedFileTypes.split(',').map((item) => {return item.trim()})
+class ImgDropAndCrop extends Component {
+    constructor(props){
+        super(props)
+        this.imagePreviewCanvasRef = React.createRef()
+        this.state = {
+            imgSrc: null,
+            crop: {
+                aspect: 1 / 1,
+            },
+        }
+    }
+
+    verifyFile = (files) => {
+        if (files && files.length > 0) {
+            const currentFile = files[0]
+            const currentFileType = currentFile.type
+            const currentFileSize = currentFile.size
+            if(currentFileSize > imageMaxSize) {
+                alert('This file is not allowed.' + currentFileSize + 'bytes is too large')
```

Comment:
The texts in alerts should be translatable. `react-intl`is used for translations in this project.